### ISSUE TITLE
Keyboard navigation through Select Component options results in lost focus/no screen reader feedback when reaching end of list

### ISSUE THREAD

**Issue by ahelminiak-rdg** [chakra-ui/ark#3470]
### Description

When I navigate via keyboard through the select component options with my screen reader on, I expected it to announce each option and maintain focus treatment on the highlighted option. Instead, when it reaches the end (or the beginning if you’re navigating up instead of down), there is a keystroke where there is no longer any highlighted option visible or screen reader announcement of what is selected, until the same arrow key is pressed again at which point highlighting resumes.

It only happens when reaching the end of the list and continuing in the same navigation direction, leading me to believe it’s related to the functionality that allows users to cycle through the list repeatedly instead of stopping at the top or bottom when it is reached.

### Link to Reproduction (or Detailed Explanation)

https://chakra-ui.com/docs/components/select

### Steps to Reproduce

1. Navigate to any select component example in the [Chakra documentation](https://chakra-ui.com/docs/components/select) or [Ark documentation](https://ark-ui.com/docs/components/select).
2. Turn on your screen reader (VoiceOver was used to test this issue).
3. Use keyboard navigation to enter select component and open the options (tab into it, press space bar to open options)
4. Use down arrow key* to navigate to the end of the options list, wait for any screen reader feedback to conclude to make sure testing is accurate
5. Press down arrow key again, highlight disappears from the entire list, screen reader makes no announcement
6. Press down arrow key one more time, highlight reappears on first option on list, screen reader announcing returns to normal.

*the same behavior occurs but in reverse if using the up arrow key to navigate backwards through the options.

### Ark UI Version

5.9.1

### Framework

- [x] React
- [ ] Solid
- [ ] Vue

### Browser

Chrome 135.0.7049.97, Firefox 138.0.3 , Safari 18.4

### Additional Information

Discovered the issue with a Chakra 3 React project, which points to Ark as the source of the Select component. 

This is an accessibility issue that fails success of at least this set of criteria, but is also an issue for screen reader issues, who would expect feedback and a particular action when repeating a keyboard navigation pattern within the same component: [WCAG 2.4.7 Focus Visible (AA)](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
